### PR TITLE
[WIP] Refactoring and testing

### PR DESCRIPTION
**Pull request by kogisu**
1. Moved files around for cleaner management
  - unit test files moved to unit directory (one file for each module test)
  - server startup and db startup connections moved to startup directory (one for each)
  - aws startup moved to services directory (aws setup still wip)
2. Added getAverageRating function
  - queries db to obtain averages of each rating
  - adds rating data to response object
3. Added rounding function to getReviewsById average rating (rounds to nearest 0.5)
4. Refactored return structure for query (returns nested objects for review, user, ratings).

### CODE REVIEW

File: example/reviews/server/server.js [hunk 1]
```diff
@@ -1,12 +0,0 @@
-const express = require('express');
```
Comment (allenhsieh) [inline example/reviews/server/server.js:1]: Hahaha good call on removing the example folder

File: server/app.js [hunk 2]
```diff
@@ -14,11 +13,10 @@ app.use(morgan('tiny'));
 app.use(express.static(__dirname + '../client/dist'));
 
 //Routing
-app.use('/reviews', reviews);
-
-const port = process.env.PORT || 3003;
-const server = app.listen(port, () => {
```
Comment (allenhsieh) [inline server/app.js]: Oh nice, it looks like you split your server into server + app too? I had to do that to get tests working

File: unit/app.test.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const supertest = require('supertest');
+const app = require('../server/app');
+const request = supertest(app);
+
+//Although db module is not used here, db needs to be mocked for function to run
+const db = require('../db');
+jest.mock('../db');
+
+describe('routing /reviews', () => {
```
Comment (allenhsieh) [inline unit/app.test.js]: nit picky, but i would do tests for every positive/negative route that you can think of

Comment (allenhsieh) [reply unit/app.test.js]: Sorry, i was unclear in my previous comments. For positive/negative i meant:

1. What should happen when someone hits `/`
2. What should happen when someone hits a `/` with request you don't want to handle (POST, OPTION, PUT, ETC)
3. Will you be hitting `/reviews/:id` as an endpoint or just `/reviews`, what should happen when you hit those end points and when do you want to happen if like, a listing `id` does not exist

File: server/app.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
+app.use(morgan('tiny'));
+app.use(express.static(__dirname + '../client/dist'));
+
+//Routing
+// app.use('/reviews', reviews);
+app.get('/reviews/:id', async(req, res) => {
+  const id = req.params.id;
+  try {
```
Comment (allenhsieh) [inline server/app.js:20]: very nice, i like how you decided to use try/catch

File: example/reviews/server/server.js (hunk 1 above)
Comment (kogisu) [reply example/reviews/server/server.js:1]: yeah !

File: server/app.js (hunk 2 above)
Comment (kogisu) [reply server/app.js]: yes, i tried running the tests as they were, but they didn't pass until i split the server/app, and the db/app.  Also removed routing entirely.

**Review by jondionowens — APPROVED** [review]
I see no issues with any of this, though I have no experience with postgreSQL.

**Pull request merged**